libraries/abstractions/pkcs11/utest/iot_pkcs11_mbedtls_utest.c

```diff
@@ -171,112 +299,540 @@ void test_pkcs11_C_InitializeInitTwice( void )
+
+/*!
+ * @brief C_GetMechanismInfo happy path.
+ *
+ */
+void test_pkcs11_C_GetMechanismInfo( void )
+{
+    CK_RV xResult = CKR_OK;
+    CK_MECHANISM_TYPE xType = CKM_RSA_PKCS;
+    CK_MECHANISM_INFO xInfo = { 0 };
 
+    xResult = C_GetMechanismInfo( 0, xType, &xInfo );
+
+    TEST_ASSERT_EQUAL( 2048, xInfo.ulMinKeySize );
+    TEST_ASSERT_EQUAL( 2048, xInfo.ulMaxKeySize );
+    TEST_ASSERT_EQUAL( CKF_SIGN, xInfo.flags );
+    TEST_ASSERT_EQUAL( CKR_OK, xResult );
+
+    xType = CKM_RSA_X_509;
+    memset( &xInfo, 0, sizeof( CK_MECHANISM_INFO ) );
+    xResult = C_GetMechanismInfo( 0, xType, &xInfo );
+
+    TEST_ASSERT_EQUAL( 2048, xInfo.ulMinKeySize );
+    TEST_ASSERT_EQUAL( 2048, xInfo.ulMaxKeySize );
+    TEST_ASSERT_EQUAL( CKF_VERIFY, xInfo.flags );
+    TEST_ASSERT_EQUAL( CKR_OK, xResult );
+
+    #if ( pkcs11configSUPPRESS_ECDSA_MECHANISM != 1 )
+        xType = CKM_ECDSA;
+        memset( &xInfo, 0, sizeof( CK_MECHANISM_INFO ) );
+        xResult = C_GetMechanismInfo( 0, xType, &xInfo );
+
+        TEST_ASSERT_EQUAL( 256, xInfo.ulMinKeySize );
+        TEST_ASSERT_EQUAL( 256, xInfo.ulMaxKeySize );
+        TEST_ASSERT_EQUAL( CKF_SIGN | CKF_VERIFY, xInfo.flags );
+        TEST_ASSERT_EQUAL( CKR_OK, xResult );
+
+        xType = CKM_EC_KEY_PAIR_GEN;
+        memset( &xInfo, 0, sizeof( CK_MECHANISM_INFO ) );
+        xResult = C_GetMechanismInfo( 0, xType, &xInfo );
+
+        TEST_ASSERT_EQUAL( 256, xInfo.ulMinKeySize );
+        TEST_ASSERT_EQUAL( 256, xInfo.ulMaxKeySize );
+        TEST_ASSERT_EQUAL( CKF_GENERATE_KEY_PAIR, xInfo.flags );
+        TEST_ASSERT_EQUAL( CKR_OK, xResult );
+    #endif /* if ( pkcs11configSUPPRESS_ECDSA_MECHANISM != 1 ) */
+
+    xType = CKM_SHA256;
+    memset( &xInfo, 0, sizeof( CK_MECHANISM_INFO ) );
+    xResult = C_GetMechanismInfo( 0, xType, &xInfo );
+
+    TEST_ASSERT_EQUAL( 0, xInfo.ulMinKeySize );
+    TEST_ASSERT_EQUAL( 0, xInfo.ulMaxKeySize );
+    TEST_ASSERT_EQUAL( CKF_DIGEST, xInfo.flags );
+    TEST_ASSERT_EQUAL( CKR_OK, xResult );
+}
+
+/*!
+ * @brief C_GetMechanismInfo NULL info pointer.
+ *
+ */
+void test_pkcs11_C_GetMechanismInfoNullInfoPtr( void )
+{
+    CK_RV xResult = CKR_OK;
+    CK_MECHANISM_TYPE xType = CKM_RSA_PKCS;
+
+    xResult = C_GetMechanismInfo( 0, xType, NULL );
     TEST_ASSERT_EQUAL( CKR_ARGUMENTS_BAD, xResult );
 }
 
-/* ======================  TESTING C_GetFunctionList  ============================ */
+/*!
+ * @brief C_GetMechanismInfo Bad Mechanism.
+ *
+ */
+void test_pkcs11_C_GetMechanismInfoBadMech( void )
+{
+    CK_RV xResult = CKR_OK;
+    CK_MECHANISM_TYPE xType = -1;
+    CK_MECHANISM_INFO xInfo = { 0 };
+
+    xResult = C_GetMechanismInfo( 0, xType, &xInfo );
+    TEST_ASSERT_EQUAL( CKR_MECHANISM_INVALID, xResult );
+}
+
+/* ======================  TESTING C_InitToken  ============================ */
 
 /*!
- * @brief C_GetFunctionList happy path.
+ * @brief C_InitToken Happy path.
+ *
+ * @note this will need to be updated if tokens are supported
```

Comment:
I think it would be more accurate to say that our port behaves like a fixed token and that it doesn't do anything when C_InitToken is called. The former is unlikely to change, but the latter might.

Reply:
Done